components/zeekopenbsm/tests/openbsmconsumer.cpp

```diff
@@ -0,0 +1,103 @@
+#include "openbsmconsumer.h"
```

Comment:
Could you include some tests here which exercise an actual `OpenbsmConsumer` instance?

Maybe it would make sense to change to a more end-to-end strategy here instead. For that you could change the constructor of `OpenbsmConsumer` so one can optionally pass in an alternative audit pipe path (defaulting to the "real" value). You could then inject events into a test pipe. That would also give us a chance to convince ourselves that we haven't introduced deadlocks, and to more exhaustively exercise the `extract*` functions where the current tests only cover a small subset.

Reply:
I can add a test for e2e testing `OpenbsmConsumer` but that requires a lot of changes. I prefer to make a separate PR for that test. Further, as of now we only have one consumer and one producer running, there are fewer chances that we introduced some deadlocks.

Reply:
We should really include a test of this functionality with this PR, currently we have almost no coverage. How did you test this?

Reply:
The workflow of producer in openbsmconsumer.cpp is as follows:
`fetchRecordsFromAuditPipe` -> `parseRecordIntoTokens` -> `populateEventFromTokens` -> `extract*`.
Now, we have tests for `populateEventFromTokens` function which basically calls all the `extract*` methods to return `Event` object. Thus, our coverage is pretty good already.
To write a test that starts from `fetchRecordsFromAuditPipe` function is not that easy because you need to make sure in this new test that whatever you write to the pipe must be readable in `parseRecordIntoTokens`  using `au_read_rec` function. I cannot write some arbitrary text into the pipe and then read it using `au_read_rec`. 

Besides that, the grunt of work is done in `populateEventFromTokens` function, which we are already testing while `fetchRecordsFromAuditPipe` and `parseRecordIntoToken` functions just read records from the pipe and then tokenize them, respectively.

Reply:
@bbannier To elaborate more on this, like all the other tests we have for Zeek Agent, we mock a log message from the underlying service (e.g., auditd, openbsm) _without_ running the service and then we have tests that checks the validity of our parsing functions. So I used the same pattern here. 
If you still think we should write a test that generates the log messages by running the actual service then could you give me some advice on how should I implement it.

Reply:
> Besides that, the grunt of work is done in `populateEventFromTokens` function, which we are already testing ...

I had looked at coverage information from running the tests, and for some reason that reported no coverage where in the code there clearly is coverage. This seems good enough to me. Sorry for the hold-up because of that.